semgrep-core/tests/rules/taint_no_builtin_props.py

```diff
@@ -0,0 +1,14 @@
+
+def bad():
+    x = "password"
+    y = x
+    z = y
+    #ruleid: secrets
+    sink(z)
+
+def ok():
+    x = "password"
+    y = x
+    z = y+"ok"
```

Comment:
This test case does not justify to me disabling all propagators for secrets. I feel like I don't have a good understanding of why that is necessary. It seems to me that we just don't want concatenation to propagate taint (though I'd also like to see a more illustrative use case for why---in this case, surely having the password as part of the string). Other methods of propagation still preserve the value of the secret whole. Could we never propagate a secret through an array?

Reply:
We're trying to understand what is needed for secrets. I see Lewis writes rules where he doesn't want "anything" than assignments to propagate taint, so we try that and see how it goes. Probably some of this leads to FNs, but that's always a trade-off you make when you want to minimize FPs. It's still possible to write explicit taint propagators.